diff --git a/driver/build/extra_symbols.rs b/driver/build/extra_symbols.rs
--- a/driver/build/extra_symbols.rs
+++ b/driver/build/extra_symbols.rs
@@ -26,7 +26,8 @@
     // smoelius: `CARGO_MANIFEST_DIR` is the directory of the `dylint_driver` library, not the
     // Dylint driver being built. We cannot easily access the Dylint driver's rust-toolchain file,
     // so we use the `RUSTUP_TOOLCHAIN` environment variable instead.
-    let rustup_toolchain = env::var(env::RUSTUP_TOOLCHAIN)?;
+    let rustup_toolchain = env::var(env::RUSTUP_TOOLCHAIN)
+        .or_else(|_| env::var(env::TARGET).map(|target| format!("nightly-{target}")))?;
     let Some(channel) = clippy_utils_has_extra_symbols(&rustup_toolchain)? else {
         write(&path_buf, "const EXTRA_SYMBOLS: &[&str] = &[];")
             .with_context(|| format!("`write` failed for `{}`", path_buf.display()))?;
diff --git a/driver/src/lib.rs b/driver/src/lib.rs
--- a/driver/src/lib.rs
+++ b/driver/src/lib.rs
@@ -398,7 +398,11 @@

 pub fn dylint_driver<T: AsRef<OsStr>>(args: &[T]) -> Result<()> {
     if args.len() <= 1 || args.iter().any(|arg| arg.as_ref() == "-V") {
-        println!("{} {}", env!("RUSTUP_TOOLCHAIN"), env!("CARGO_PKG_VERSION"));
+        println!(
+            "{} {}",
+            option_env!("RUSTUP_TOOLCHAIN").unwrap_or("nightly"),
+            env!("CARGO_PKG_VERSION"),
+        );
         return Ok(());
     }
